client/src/components/Related/ProductCard.jsx

```diff
@@ -1,42 +1,62 @@
 import React, { useState, useEffect } from 'react';
 import ratingsAPI from '../../API/Ratings.js';
+import { useNavigate } from 'react-router-dom';
 
 const RelatedProduct = (props) => {
-  const [averageRating, setAverageRating] = useState(0);
+  const navigate = useNavigate();
+  const [averageRating, setAverageRating] = useState();
+  const stars = props.generateStars(averageRating, 'related');
 
+  useEffect(() => {
+    ratingsAPI.getReviewMetadata(props.product.id)
+      .then((metadata) => {
+        setAverageRating(getAverageRating(metadata.ratings));
+        props.setIsFetching(false);
+      });
+  }, []);
+
+  const routeChange = () => {
+    const path = `/${props.product.id.toString()}`;
+    navigate(path);
+  };
+
+  const displayComparison = () => {
+    return;
+  };
```

Comment:
Not sure if you're planning on doing something with this later, but if not, I would delete this.